docker/upgrade.pp

```diff
@@ -0,0 +1,12 @@
+node default {
+  class { '::puppet_agent':
+    package_version => $facts['to_version'],
+    # Upgrades in Docker cannot start the puppet service due to systemd
```

Comment:
beaker-docker makes systemd work by using `/sbin/init` as the `CMD`. I'm not sure if it's possible to successfully work that in here, but perhaps worth a try. A huge caveat for that, though, is that the container must run in privileged mode for it to work. This can be particularly scary when there's no docker machine in between your laptop and the container (see https://github.com/puppetlabs/beaker-hostgenerator/pull/138 for an example problem that can arise if you're not careful)

Reply:
I took a look at `/sbin/init` and even using `centos/systemd` as the base container, but `/sbin/init` is mostly for starting services by name and I couldn't find a way to run the upgrade within that context.